sourced/ml/models/id_splitter.py

```diff
@@ -0,0 +1,124 @@
+import logging
+import os
+import string
+from typing import List, Sequence, Tuple
+
+import keras
+from keras import backend
+from keras.preprocessing.sequence import pad_sequences
+from modelforge import Model, register_model
+import numpy
+
+from sourced.ml.algorithms.id_splitter.nn_model import (f1score, precision,
+                                                        recall)
+from sourced.ml.models.license import DEFAULT_LICENSE
+
+
+@register_model
+class IdentifierSplitterNN(Model):
+    """
+    Bi-Directionnal LSTM Model. Splits identifiers without need for a conventional pattern.
+    """
+    NAME = "id_splitter_nn"
+    VENDOR = "source{d}"
+    DESCRIPTION = "Model that contains source code identifier splitter BiLSTM weights."
+    LICENSE = DEFAULT_LICENSE
+
+    DEFAULT_MAXLEN = 40
+    DEFAULT_PADDING = "post"
+    DEFAULT_MAPPING = None
+
+    def construct(self, model: "keras.models.Model" = None,
+                  session: "tf.Session" = None,
+                  maxlen: int = DEFAULT_MAXLEN,
+                  padding: str = DEFAULT_PADDING,
+                  mapping: dict = DEFAULT_MAPPING) -> "IdentifierSplitterNN":
+        assert model is not None
+
+        if session is not None:
+            backend.tensorflow_backend.set_session(session)
+        os.environ["TF_CPP_MIN_LOG_LEVEL"] = "1"
```

Comment:
I do not think we should set it here because it is unobvious for the user. If it is required, the user can set it by himself